### ISSUE TITLE
Franka Hand Binary Not found

### ISSUE THREAD

**Issue by ahmeda14960** [facebookresearch/fairo#1373]
## Type of Issue

Select the type of issue:
- [x] Bug report (to report a bug)
- [ ] Feature request (to request an additional feature)
- [ ] Tracker (I am just using this as a tracker)
- [ ] Refactor request
- [ ] Documentation Ask

## Description
First off, thank you all for a well built overall repo!

I've gotten polymetis to work with a Franka Panda Robot I am using, but I cannot use the Franka Gripper. when I follow the directions of the local install, and then simply run `launch_gripper.py gripper=franka_hand` I am told the franka_hand_client binary cannot be found, despite the fact that building libfranka and polymetis succeeded. 
## Current Behavior
running  `launch_gripper.py gripper=franka_hand` 
leads to following error
```
Warning: Failed to load 'libtorchscript_pinocchio.so' from CONDA_PREFIX, loading from default build directory instead: '/home/panda5/fairo/polymetis/polymetis/build/torch_isolation/libtorchscript_pinocchio.so'
Warning: Failed to load 'libtorchrot.so' from CONDA_PREFIX, loading from default build directory instead: '/home/panda5/fairo/polymetis/polymetis/build/torch_isolation/libtorchrot.so'
/home/panda5/anaconda3/envs/polymetis-local/lib/python3.8/site-packages/hydra/plugins/config_source.py:190: UserWarning: 
Missing @package directive gripper/franka_hand.yaml in pkg://polymetis.conf.
See https://hydra.cc/docs/next/upgrades/0.11_to_1.0/adding_a_package_directive
  warnings.warn(message=msg, category=UserWarning)
Gripper server running at 0.0.0.0:50052.
[2022-10-03 19:18:56,228][polymetis.robot_client.executable_robot_client][INFO] - === Config: ===
exec: franka_hand_client
robot_ip: 172.16.0.2
control_ip: 0.0.0.0
control_port: 50052
robot_client_metadata_path: /tmp/tmphrzoo0zg

Traceback (most recent call last):
  File "/home/panda5/fairo/polymetis/polymetis/python/scripts/launch_gripper.py", line 40, in main
    gripper_client.run()
  File "/home/panda5/fairo/polymetis/polymetis/python/polymetis/robot_client/executable_robot_client.py", line 69, in run
    assert path_to_exec, f"Unable to find binary {self.executable_cfg.exec}"
AssertionError: Unable to find binary franka_hand_client

```

## Expected Behavior

for the server to start without issues
## Steps to reproduce

I created a clean conda env, directly following all the steps for the local installation here:  https://facebookresearch.github.io/fairo/polymetis/installation.html. The only thing I changed was when running the install libfranka script, I ran `./scripts/build_libfranka.sh 0.8.0` because this is the version that happens to work with my NUC /RT_kernel spec

**Comment by 1heart** [comment]
Hmm, just to check -- can you try running the following to rebuild? (Mostly making sure BUILD_FRANKA is ON, which is the flag that adds the [franka_hand_client](https://github.com/facebookresearch/fairo/blob/main/polymetis/polymetis/CMakeLists.txt#L203) )

```bash
cd ./polymetis/build
rm -rf ./*
cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_FRANKA=ON
make -j
```

**Comment by ahmeda14960** [comment]
Hello,

So it looks like when running 
` cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_FRANKA=ON `
That I recieve some warnings which I think explain the issue 

```
Found Protobuf: /usr/lib/x86_64-linux-gnu/libprotobuf.so;-lpthread (found version "3.0.0") 
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.1.1")  
-- Found c-ares: /home/panda5/anaconda3/envs/polymetis-local/lib/cmake/c-ares/c-ares-config.cmake (found version "1.18.1") 
-- Found RE2 via CMake.
Compiling Franka client with libfranka from src/clients/franka_panda_client/third_party/libfranka/build
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Configuring done
CMake Warning at CMakeLists.txt:124 (add_executable):
  Cannot generate a safe runtime search path for target run_server because
  files in some directories may conflict with libraries in implicit
  directories:

    runtime library [libz.so.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib
    runtime library [libssl.so.1.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib
    runtime library [libcrypto.so.1.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib

  Some of these libraries may not be found correctly.


CMake Warning at CMakeLists.txt:128 (add_executable):
  Cannot generate a safe runtime search path for target
  empty_statistics_client because files in some directories may conflict with
  libraries in implicit directories:

    runtime library [libz.so.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib
    runtime library [libssl.so.1.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib
    runtime library [libcrypto.so.1.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib

  Some of these libraries may not be found correctly.


CMake Warning at CMakeLists.txt:193 (add_executable):
  Cannot generate a safe runtime search path for target franka_panda_client
  because files in some directories may conflict with libraries in implicit
  directories:

    runtime library [libz.so.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib
    runtime library [libssl.so.1.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib
    runtime library [libcrypto.so.1.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib

  Some of these libraries may not be found correctly.


CMake Warning at CMakeLists.txt:203 (add_executable):
  Cannot generate a safe runtime search path for target franka_hand_client
  because files in some directories may conflict with libraries in implicit
  directories:

    runtime library [libz.so.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib
    runtime library [libssl.so.1.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib
    runtime library [libcrypto.so.1.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/panda5/anaconda3/envs/polymetis-local/lib

  Some of these libraries may not be found correctly.


-- Generating done
-- Build files have been written to: /home/panda5/fairo/polymetis/polymetis/build
```

As you can see above it seems like it's not able to create the search  path for the franka hand client, but the weird thing is it complains that some runtime libraries are hidden to polymetis which shouldn't be the case as they're in ` /usr/lib/x86_64-linux-gnu ` which isn't a hidden directory? FWIW running `make -j` seems to be working correctly:

```
[  4%] Generating polymetis.pb.cc, polymetis.pb.h, polymetis.grpc.pb.cc, polymetis.grpc.pb.h
[ 13%] Building CXX object torch_isolation/CMakeFiles/torchrot.dir/src/rotations.cpp.o
[ 13%] Building CXX object torch_isolation/pinocchio_isolation/CMakeFiles/pinocchio_wrapper.dir/src/pinocchio_wrapper.cpp.o
[ 18%] Building CXX object torch_isolation/CMakeFiles/torch_server_ops.dir/src/torch_server_ops.cpp.o
[ 27%] Building CXX object CMakeFiles/generated_grpc_protos.dir/polymetis.grpc.pb.cc.o
[ 27%] Building CXX object CMakeFiles/generated_grpc_protos.dir/polymetis.pb.cc.o
[ 31%] Linking CXX static library libgenerated_grpc_protos.a
[ 31%] Built target generated_grpc_protos
[ 36%] Building CXX object CMakeFiles/empty_statistics_client.dir/src/clients/empty_statistics_client.cpp.o
[ 40%] Building CXX object CMakeFiles/franka_panda_client.dir/src/clients/franka_panda_client/franka_panda_client.cpp.o
[ 45%] Building CXX object CMakeFiles/franka_hand_client.dir/src/clients/franka_panda_client/franka_hand_client.cpp.o
[ 50%] Linking CXX executable franka_hand_client
[ 54%] Linking CXX executable empty_statistics_client
[ 54%] Built target franka_hand_client
[ 54%] Built target empty_statistics_client
[ 59%] Linking CXX executable franka_panda_client
[ 59%] Built target franka_panda_client
[ 63%] Linking CXX shared library libtorch_server_ops.so
[ 63%] Built target torch_server_ops
[ 68%] Linking CXX shared library libtorchrot.so
[ 68%] Built target torchrot
[ 72%] Linking CXX shared library libpinocchio_wrapper.so
[ 72%] Built target pinocchio_wrapper
[ 77%] Building CXX object torch_isolation/CMakeFiles/torchscript_pinocchio.dir/src/pinocchio.cpp.o
[ 81%] Linking CXX shared library libtorchscript_pinocchio.so
[ 81%] Built target torchscript_pinocchio
[ 86%] Building CXX object CMakeFiles/polymetis_server.dir/src/polymetis_server.cpp.o
[ 90%] Linking CXX static library libpolymetis_server.a
[ 90%] Built target polymetis_server
[ 95%] Building CXX object CMakeFiles/run_server.dir/src/run_server.cpp.o
[100%] Linking CXX executable run_server
[100%] Built target run_server
```

**Comment by 1heart** [comment]
It's saying `/home/panda5/anaconda3/envs/polymetis-local/lib` libraries are found before the system-level ones, which is the intended behavior of Conda anyways, so that should be totally fine. Can you check that the `franka_panda_client` executable exists in the build directory tree?

**Comment by ahmeda14960** [comment]
I see... the executable does exist:

```
~/fairo/polymetis/polymetis/build$ ls
CMakeCache.txt           libgenerated_grpc_protos.a  polymetis_pb2.py
CMakeFiles               libpolymetis_server.a       polymetis.pb.cc
cmake_install.cmake      Makefile                    polymetis.pb.h
empty_statistics_client  polymetis.grpc.pb.cc        run_server
franka_hand_client       polymetis.grpc.pb.h         torch_isolation
franka_panda_client      polymetis_pb2_grpc.py
(base) panda5@panda5-NUC11PAHi7:~/fairo/polymetis/polymetis/build$ test -x franka_hand_client && echo true || echo false
true
```

**Comment by 1heart** [comment]
Do you mean that even after `franka_hand_client` exists in the build directory, you are still getting the same `AssertionError: Unable to find binary franka_hand_client` error trying to run `launch_gripper.py gripper=franka_hand`?

**Comment by 1heart** [comment]
Sorry, you're right -- I can reproduce this issue on our end and have fixed it in #1375. Please pull `main` branch after that PR is merged.

**Comment by ahmeda14960** [comment]
Great, thank you!
